### ISSUE TITLE
Add integration tests

### ISSUE THREAD

**Issue by jameslamb** [bgruening/conda_r_skeleton_helper#43]
Right now this repository doesn't have any tests to catch incompatibilities with its upstream dependencies (`conda-build`, mainly). I think it would be valuable to set up some automated testing on the code to improve confidence in the new releases and to make  the project more friendly to outside contributors.

One set of tests could be linting (`flake8` / `pycodestyle` for Python, `lintr` for R) to catch minor style and code-quality issues like unused imports.

Another could involve a process that, for some hard-coded list of packages on CRAN, runs  `run.R` and / or `run.py` and then checks the produced recipes with `conda-smithy lint` or something similar.

Given these tests, it would be useful to set up Travis CI or similar to run them on every change to the repo.

**Comment by jdblischak** [comment]
@jameslamb Thanks for raising this Issue! In general I think regular tests with CI are great, but I worry about this particular case. These scripts are ad-hoc and mostly involve crude line-by-line string substitutions. Thus on the one hand I don't think it is worth investing the engineering time to add testing and CI infrastructure. On the other hand, due to limited time, these scripts have become the recommended method for generating conda-forge R recipes, and thus we need to ensure that they work.

Also I've long believed that a better long-term solution would be to directly update `conda skeleton` to have a flag such as `--conda-forge`, which would implement the conda-forge preferred formatting. Directly editing the Python script is much more dependable than find/replace with regexes, and it also allows users to directly run `conda skeleton`. And if a `--conda-forge` flag is considered by the maintainers to be too difficult to implement, we could always try smaller flags such as `--no-comments`.

Some past examples of directly editing the CRAN skeleton to avoid having to handle it downstream: https://github.com/conda/conda-build/pull/3284, #32, https://github.com/conda/conda-build/pull/3566

Thus if you or any other interested developer has time to invest in the infrastructure for building R recipes for conda-forge, I'd recommend first trying to make this repository obsolete by making more contributions to the upstream recipe.

Also complicating the situation is that fact that `conda skeleton` is a moving target. A [replacement](https://github.com/marcelotrevisani/grayskull) is in the works:

> The Grayskull project was created with the intention to eventually replace the conda skeleton. The main goal of this project is to generate concise recipes for conda-forge.
>
> The current status of grayskull is generating recipes just looking for PyPI, but in the future we will expand that to also support to load recipes and also generate recipes looking for other repositories, such as, CRAN, Conan, etc..

AFAIK there is still not yet [support for R recipes](https://github.com/marcelotrevisani/grayskull/issues/7). So this replacement is another place that could use some attention.

In summary, I am open to you adding some automating testing infrastructure to this repository, but I also want to make sure that you don't spend too much time on it. Thoughts?

**Comment by jdblischak** [comment]
Coming back to this issue after a year, I am more confident that the best course of action is to focus our development efforts on improving the upstream CRAN skeleton in conda-build. It already has automated testing and CI setup. To me it makes more sense to improve the existing infrastructure than to create a secondary one that we would have to maintain ourselves.

Recently I've:

* Submitted PR https://github.com/conda/conda-build/pull/4302 to add a flag `--no-comments`
* Submitted PR https://github.com/conda/conda-build/pull/4317 to refactor the existing CRAN skeleton tests and add new tests

My plan is to reduce the helper script features to the point that it is no longer required, which I've started in PR #53.

If anyone has the bandwidth to contribute to improving the conda-forge R infrastructure, please reach out. We need to upstream more of our edits and also add more tests to conda-build.
